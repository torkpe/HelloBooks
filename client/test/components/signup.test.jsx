import React from 'react';
import toJson from 'enzyme-to-json';
import sinon from 'sinon';
import { configure, shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import render from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';

import storeConfiguration from '../../store/store';
import connectedSignup, { Signup } from '../../components/Signup.jsx';

configure({ adapter: new Adapter() });

describe('<SignUp />', () => {
  const props = {
    userSignup: jest.fn(),
    clearSignupState: jest.fn(),
    signup: {
      isLoading: false,
      successfullySignedup: {
        message: 'mvhbvhjbbbb',
      },
      errors: {
        message: 'fdghjk'
      }
    },
    successfullySignedup: {
      message: 'mvhbvhjbbbb',
    },
  };
  let event = {
    preventDefault: jest.fn(),
    target: {
      username: 'okon',
      password: 'abcde',
      email: 'okon@yahoo.com',
    }
  };
  const mockStore = configureMockStore();
  let mountedComponent;
  let store;
  let wrapper;
  let shallowComponent;
  beforeEach(() => {
    store = mockStore(storeConfiguration());
    mountedComponent = mount(<Signup {...props} />);
    shallowComponent = shallow(<Signup {...props} />);
    wrapper = mount(<Provider store={storeConfiguration()}><connectedSignup /></Provider>);
  });
  it('renders <SignUp /> component', () => {
    const tree = toJson(mountedComponent);
    expect(tree).toMatchSnapshot();
  });
  it('calls handleChange event', () => {
    sinon.spy(Signup.prototype, 'onChange');
    shallowComponent.instance().onChange(event);
    expect(Signup.prototype.onChange.calledOnce).toEqual(false);
  });
  it('calls handle signup event after signup button is clicked', () => {
    sinon.spy(Signup.prototype, 'onSubmit');
    props.successfullySigned = {
      message: 'mvhbvhjbbbb'
    };
    event = {
      preventDefault: jest.fn()
    };
    shallowComponent = shallow(<Signup {...props} />);
    shallowComponent.instance().onSubmit(event);
    expect(Signup.prototype.onSubmit.calledOnce).toEqual(true);
  });
  it('calls componentWillReceiveProps', () => {
    sinon.spy(Signup.prototype, 'componentWillReceiveProps');
    shallowComponent.instance().componentWillReceiveProps(props);
    expect(Signup.prototype.componentWillReceiveProps.calledOnce).toEqual(true);
  });
  it('calls componentWillReceiveProps when there is no error message', () => {
    props.signup.errors.message = null;
    shallowComponent.instance().componentWillReceiveProps(props);
    expect(Signup.prototype.componentWillReceiveProps.calledOnce).toEqual(false);
  });
});
