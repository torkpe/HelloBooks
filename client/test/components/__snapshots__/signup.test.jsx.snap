// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignUp /> renders <SignUp /> component 1`] = `
<Signup
  clearSignupState={[MockFunction]}
  signup={
    Object {
      "errors": Object {
        "message": "fdghjk",
      },
      "isLoading": false,
      "successfullySignedup": Object {
        "message": "mvhbvhjbbbb",
      },
    }
  }
  successfullySignedup={
    Object {
      "message": "mvhbvhjbbbb",
    }
  }
  userSignup={[MockFunction]}
>
  <div
    className="mdl-grid"
  >
    <div
      className="contents"
    >
      <div
        className="card-enlarge form-card mdl-card mdl-shadow--3dp"
      >
        <form
          onSubmit={[Function]}
        >
          <h5>
            Signup
          </h5>
          <EmailForm
            onChange={[Function]}
          >
            <div
              className="card-content card-wrapper input-wrapper"
            >
              <input
                id="email"
                name="email"
                onChange={[Function]}
                placeholder="Email"
                type="email"
              />
            </div>
          </EmailForm>
          <button
            className="
    mdl-button
    mdl-js-button
    mdl-button--raised
    mdl-button--colored
    "
            disabled={false}
            id="button1"
            name="signup"
          >
            Sign up
          </button>
        </form>
        <div
          className="mdl-card__supporting-text"
        >
          Already have an account? Sign in below
        </div>
        <Link
          onlyActiveOnIndex={false}
          style={Object {}}
          to="/signin"
        />
      </div>
    </div>
  </div>
</Signup>
`;
